### DISCUSSION TITLE
undeclared identifier: 'AddLoanDialogState'

### DISCUSSION THREAD

**Discussion by ITwrx** [can-lehmann/owlkettle#46] [Q&A] [answered]
hi,
I have a `custom_dialog` that opens and seems to work fine, but i can't save the entered data, because i can't run my proc (`addNewLoanToLoans()`), because `AddLoanDialogState` is not available for some reason. I'm trying to follow the owlkettle `custom_dialog` example like in [ln 102](https://github.com/can-lehmann/owlkettle/blob/main/examples/dialogs/custom_dialog.nim): `app.users.add(UserDialogState(state).user)` and  assumed `AddLoanDialogState` would be available anywhere `AddLoanDialog` is available, but i must be missing something. The file that contains `AddLoanDialog` is imported to the file that contains the button code below (as i say, it opens fine). I can show any code you want to see. Just refraining unless necessary, to keep the Discussion clean and out of laziness, as i may have to "curate" it prior to posting.

Here is my button code:

```
Button {.expand: false.}:
  text = "+ Loan"
  style = {ButtonSuggested}
  proc clicked() =
    let (res, state) = loanListView.open(gui(AddLoanDialog()))
    if res.kind == DialogAccept:
      addNewLoanToLoans(AddLoanDialogState(state).newLoan) 
```
thanks

**Comment by can-lehmann** [comment, marked as answer]
I assume that you have a line like `export AddLoanDialog` in the module that contains the definition for the dialog. In order for `AddLoanDialogState` to be available in other modules, you need to export it as well (so `export AddLoanDialog, AddLoanDialogState`).

**Reply by ITwrx** [reply]
Thanks! That did the trick. It's embarrassingly obvious now. i was kind of assuming the state type just went along for the ride with the `AddLoanDialog` type, i guess. :) I completely forgot about the `export` statement and just kept ignoring it while troubleshooting...
